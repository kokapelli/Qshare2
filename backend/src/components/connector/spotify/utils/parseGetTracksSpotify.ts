import { QShareTrack, QShareArtist } from "../../../music/musicTypes";

interface SpotifyTrack {
    name: string;
    uri: string;
    artists: SpotifyArtist[];
    album: SpotifyAlbum;
    popularity: number;
}

interface SpotifyArtist {
    uri: string;
    name: string;
}

interface SpotifyAlbum {
    images: SpotifyAlbumImage[];
}

interface SpotifyAlbumImage {
    width: number;
    height: number;
    url: string;
}

interface SpotifyGetTracksResponse {
    tracks: {
        name: string
        items: SpotifyTrack[]
        limit: number
    };
}

export interface QShareGetTrackResponse extends QShareTrack {
    popularity: number;
}

export interface QShareGetTracksResponse {
    limit: number;
    tracks: QShareGetTrackResponse[];
}

export const parseGetTracksSpotify = (data: SpotifyGetTracksResponse) => {
    const response: QShareGetTracksResponse = {} as any;
    const tracks: QShareGetTrackResponse[] = [];

    response.tracks = tracks;
    response.limit = data.tracks.limit;
    for (const track of Array.from(data.tracks.items)) {
        const newTrack: QShareGetTrackResponse = {} as any;
        newTrack.name = track.name;

        const newArtists: QShareArtist[] = [];
        for (const artist of track.artists) {
            const newArtist: SpotifyArtist = {
                uri: artist.uri,
                name: artist.name
            };

            newArtists.push(newArtist);
        }

        newTrack.artists = newArtists;

        const newImages = [];
        for (const image of track.album.images) {
            const newImage: SpotifyAlbumImage = {
                height: image.height,
                width: image.width,
                url: image.url
            };
            newImages.push(newImage);
        }

        newTrack.images = newImages;

        newTrack.uri = track.uri;
        newTrack.popularity = track.popularity;
        tracks.push(newTrack);
    }

    return response;
};
