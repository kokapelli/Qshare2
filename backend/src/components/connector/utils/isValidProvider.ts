import { SPOTIFY_PROVIDER } from "../../constants/providers";

const isValidProvider = (provider: string) => {
    return provider === SPOTIFY_PROVIDER;
};

export { isValidProvider };