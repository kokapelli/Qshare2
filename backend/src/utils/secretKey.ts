import crypto from "crypto";

const SECRET_KEY = crypto.randomBytes(256);

export default SECRET_KEY;
