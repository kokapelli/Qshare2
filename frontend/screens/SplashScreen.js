import React, { Component } from 'react'
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native'
import SnackBar from 'react-native-snackbar-component'
import { connect } from 'react-redux'
import { clientLogin } from '../actions/login'

export class SplashScreen extends Component {
  state = {
    showSnackError: false
  }

  static navigationOptions = {
    header: null
  }

  componentDidMount () {
    this.props.clientLogin()
    if (this.props.navigation.getParam('error', null)) {
      this.setState({
        showSnackError: true
      })
      setTimeout(
        () =>
          this.setState({
            showSnackError: false
          }),
        6000
      )
    }
  }

  render () {
    const error = this.props.navigation.getParam('error', null)
    const {navigate} = this.props.navigation

    return (
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Text
            style={styles.headerText}>
            QShare
          </Text>
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            testID='host'
            style={styles.button}
            onPress={() => navigate('Login')}
          >
            <Text style={styles.buttonText}> Create a Room </Text>
          </TouchableOpacity>
          <TouchableOpacity
            testID='guest'
            style={styles.button}
            onPress={() => navigate('Scanner')}
          >
            <Text style={styles.buttonText}>
              {' '}
              Join an Existing Room{' '}
            </Text>
          </TouchableOpacity>
        </View>
        <View style={styles.founderContainer}>
          <Text style={styles.founderText}>
            {' '}
            Created for PSS 2019, Uppsala University{' '}
          </Text>
        </View>
        <SnackBar
          visible={this.state.showSnackError}
          textMessage={error}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#181818'
  },
  headerContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  headerText: {
    fontSize: 80,
    textAlign: 'center',
    margin: 10,
    color: 'white'
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center'
  },
  button: {
    borderWidth: 1,
    margin: 5,
    borderColor: '#1df954',
    alignItems: 'center',
    justifyContent: 'center',
    width: 250,
    height: 70,
    backgroundColor: '#1db954',
    borderRadius: 100
  },
  buttonText: {
    color: 'white',
    fontSize: 20
  },
  founderContainer: {
    alignItems: 'center',
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    paddingBottom: 10
  },
  founderText: {
    color: 'white',
    fontSize: 15
  }
})

export default connect(null, {clientLogin})(SplashScreen)
