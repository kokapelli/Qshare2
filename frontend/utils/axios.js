import Axios from "axios"
import { settings } from "../settings/settings";

const axios = Axios.create()
axios.defaults.headers.common['Provider'] = settings.provider

export default axios