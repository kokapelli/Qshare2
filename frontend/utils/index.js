export const findByTestAttr = (component, attr)  => {
    const wrapper = component.find(`[testID='${attr}']`);
    return wrapper
  }
  