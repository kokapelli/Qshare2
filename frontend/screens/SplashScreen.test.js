import {SplashScreen} from './SplashScreen'
import { shallow, configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { findByTestAttr } from '../utils/'

//TODO Test so that navigation works onClick
//Required for jest to work
const navigation = { getParam: jest.fn((x,y) => null) };
const clientLogin = jest.fn()

configure({ adapter: new Adapter() });

describe('SplashScreen Snapshot', () => {

    it('matches the snapshot', () => {
        const tree = renderer.create(
        <SplashScreen clientLogin={clientLogin} navigation={navigation}/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

describe('SplashScreen Units', () => {

    it('component renders properly', () => {
        const rendered = renderer.create(<SplashScreen clientLogin={clientLogin} navigation={navigation}/>).toJSON()
        expect(rendered).toBeTruthy()
    });

    it('should press Hosting button', () => {
        const component = shallow(<SplashScreen clientLogin={clientLogin} navigation={navigation}/>)
        const touchable = findByTestAttr(component, 'host')
        expect(touchable.exists()).toEqual(true) 
        expect(touchable.simulate('click')).toBeDefined()
    })

    it('should press guest button', () => {
        const component = shallow(<SplashScreen clientLogin={clientLogin} navigation={navigation}/>)
        const touchable = findByTestAttr(component, 'guest')
        expect(touchable.exists()).toEqual(true) 
        expect(touchable.simulate('click')).toBeDefined()
    })
})