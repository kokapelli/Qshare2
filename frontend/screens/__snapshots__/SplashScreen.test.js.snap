// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen Snapshot matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#181818",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 80,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      QShare
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1db954",
          "borderColor": "#1df954",
          "borderRadius": 100,
          "borderWidth": 1,
          "height": 70,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 250,
        }
      }
      testID="host"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
         Create a Room 
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1db954",
          "borderColor": "#1df954",
          "borderRadius": 100,
          "borderWidth": 1,
          "height": 70,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
          "width": 250,
        }
      }
      testID="guest"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
         
        Join an Existing Room
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "left": 0,
        "paddingBottom": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 15,
        }
      }
    >
       
      Created for PSS 2019, Uppsala University
       
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#484848",
        "bottom": 0,
        "height": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "zIndex": 9999,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#484848",
          "bottom": -9999,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 14,
              "paddingBottom": 14,
              "paddingLeft": 20,
              "paddingTop": 14,
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
